operator/src/main/java/org/keycloak/operator/crds/v2alpha1/CRDUtils.java

```diff
@@ -124,4 +128,23 @@ public static Optional<Boolean> fetchIsRecreateUpdate(StatefulSet statefulSet) {
     public static Optional<String> findUpdateReason(StatefulSet statefulSet) {
         return Optional.ofNullable(statefulSet.getMetadata().getAnnotations().get(Constants.KEYCLOAK_UPDATE_REASON_ANNOTATION));
     }
+
+    public static String getRevision(Context<Keycloak> context, Keycloak keycloak) {
+        return UpdateSpec.getRevision(keycloak)
+                .orElse(defaultExplicitStrategyRevision(context));
+    }
+
+    public static String defaultExplicitStrategyRevision(Context<Keycloak> context) {
```

Comment:
I still have a concern that this will be complicated to document https://github.com/keycloak/keycloak/pull/37295#issuecomment-2663251325

Can we start off with having the revision be required via validation rule?